samples/openapi3/client/petstore/java/jersey2-java8/src/test/java/org/openapitools/client/ApiClientTest.java

```diff
@@ -1,29 +1,81 @@
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.*;
+import java.net.URI;
 import org.junit.*;
+import org.tomitribe.auth.signatures.Algorithm;
+import org.tomitribe.auth.signatures.Signer;
+import org.tomitribe.auth.signatures.SigningAlgorithm;
+import java.security.spec.PSSParameterSpec;
+import java.security.spec.MGF1ParameterSpec;
+import org.tomitribe.auth.signatures.*;
+import java.io.ByteArrayInputStream;
+import java.security.PublicKey;
+import java.security.PrivateKey;
+
 import static org.junit.Assert.*;
 
 public class ApiClientTest {
     ApiClient apiClient = null;
     Pet pet = null;
 
+    private final String privateKeyPem = "-----BEGIN RSA PRIVATE KEY-----\n" +
+            "MIICXgIBAAKBgQDCFENGw33yGihy92pDjZQhl0C36rPJj+CvfSC8+q28hxA161QF\n" +
+            "NUd13wuCTUcq0Qd2qsBe/2hFyc2DCJJg0h1L78+6Z4UMR7EOcpfdUE9Hf3m/hs+F\n" +
+            "UR45uBJeDK1HSFHD8bHKD6kv8FPGfJTotc+2xjJwoYi+1hqp1fIekaxsyQIDAQAB\n" +
+            "AoGBAJR8ZkCUvx5kzv+utdl7T5MnordT1TvoXXJGXK7ZZ+UuvMNUCdN2QPc4sBiA\n" +
+            "QWvLw1cSKt5DsKZ8UETpYPy8pPYnnDEz2dDYiaew9+xEpubyeW2oH4Zx71wqBtOK\n" +
+            "kqwrXa/pzdpiucRRjk6vE6YY7EBBs/g7uanVpGibOVAEsqH1AkEA7DkjVH28WDUg\n" +
+            "f1nqvfn2Kj6CT7nIcE3jGJsZZ7zlZmBmHFDONMLUrXR/Zm3pR5m0tCmBqa5RK95u\n" +
+            "412jt1dPIwJBANJT3v8pnkth48bQo/fKel6uEYyboRtA5/uHuHkZ6FQF7OUkGogc\n" +
+            "mSJluOdc5t6hI1VsLn0QZEjQZMEOWr+wKSMCQQCC4kXJEsHAve77oP6HtG/IiEn7\n" +
+            "kpyUXRNvFsDE0czpJJBvL/aRFUJxuRK91jhjC68sA7NsKMGg5OXb5I5Jj36xAkEA\n" +
+            "gIT7aFOYBFwGgQAQkWNKLvySgKbAZRTeLBacpHMuQdl1DfdntvAyqpAZ0lY0RKmW\n" +
+            "G6aFKaqQfOXKCyWoUiVknQJAXrlgySFci/2ueKlIE1QqIiLSZ8V8OlpFLRnb1pzI\n" +
+            "7U1yQXnTAEFYM560yJlzUpOb1V4cScGd365tiSMvxLOvTA==\n" +
+            "-----END RSA PRIVATE KEY-----\n";
+
     @Before
     public void setup() {
         apiClient = new ApiClient();
         pet = new Pet();
     }
 
+    @Test
+    public void testUpdateParamsForAuth() throws Exception {
+        Map<String, String> headerParams = new HashMap<String, String>();
+        List<Pair> queryParams = new ArrayList<>();
+
+        URI uri = new URI("/api/v1/telemetry/TimeSeries");
+
+        // auth name
+        String[] authNames = {"http_signature_test"};
+
+        HashMap<String, Authentication> authMap = new HashMap<String, Authentication>();
+
+        HttpSignatureAuth signatureAuth = new HttpSignatureAuth("some-key-1", SigningAlgorithm.HS2019, Algorithm.RSA_SHA512, null,
+                null, Arrays.asList(new String[] { "(request-target)" }), 128L);
+
+        signatureAuth.setPrivateKey(PEM.readPrivateKey(new ByteArrayInputStream(privateKeyPem.getBytes())));
+
+        authMap.put("http_signature_test", signatureAuth);
+
+        ApiClient client = new ApiClient(authMap);
+
+        client.updateParamsForAuth(authNames, queryParams, headerParams, null, null, "post", uri);
+
+        // hard to test as expire will always be different
```

Comment:
I spent far too long yesterday thinking about exactly this test.  I think this may be sufficient:

```
        Signature requestSignature = Signature.fromString(headerParams.get("Authorization"), Algorithm.RSA_SHA512);
        Verifier verify = new Verifier(PEM.readPublicKey(new ByteArrayInputStream(publicKeyPem.getBytes())), requestSignature);
        assert verify.verify("post", uri.toString(), headerParams);
```

Reply:
Thanks. Let me try.

Reply:
@timgclark I've merged this PR into master. I wonder if you can file a separate PR to enhance the test. Thank you.